p4-16/spec/P4-16-spec.mdk

```diff
@@ -2420,12 +2420,15 @@ infinite-precision integer, without a width specified.
 | `header`       | error     | allowed |  allowed |
 | header stack   | error     | error   |  allowed |
 | `header_union` | error     | error   |  allowed |
-| `struct`       | error     | error   |  allowed |
+| `struct`       | allowed[^struct_header]   | error   |  allowed |
 | `tuple`        | error     | error   |  allowed |
 |----------------|-----------|---------|----------|
 
 [^enum_header]: an `enum` type used as a field in a `header` must specify a
     underlying type and representation for `enum` elements.
+[^struct_header]: a `struct` type used as a field in a `header` must contain
+    only a bit-field, an integer, a Serializable Enums, or a Boolean.
```

Comment:
Do you mean values of type `bit<W>` when you say "bit-field", and values of type `int<W> when you say "integer"?  If so, I would recommend using `bit<W>` and `int<W>` instead of those words, because at least the term "integer" could be confused to also include unlimited precision integers.

Reply:
Will make the change to bit<W> and int<W>